devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy

```diff
@@ -0,0 +1,326 @@
+	def setpoint = device.latestValue("heatingSetpoint")
+	if (setpoint < maxHeatingSetpointTemperature) {
+		setpoint = setpoint + 1
+	}
+	setHeatingSetpoint(setpoint)
+}
+
+def setThermostatSetpointDown() {
+	def setpoint = device.latestValue("heatingSetpoint")
+	if (setpoint > minHeatingSetpointTemperature) {
+		setpoint = setpoint - 1
+	}
+	setHeatingSetpoint(setpoint)
+}
+
+def refresh() {
+	def cmds = [
+			secureEncap(zwave.batteryV1.batteryGet(), 1),
+			secureEncap(zwave.batteryV1.batteryGet(), 2),
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: 1), 1),
+			secureEncap(zwave.thermostatModeV2.thermostatModeGet(),1),
+			secureEncap(zwave.sensorMultilevelV5.sensorMultilevelGet(), 2),
+			secureEncap(zwave.configurationV1.configurationGet(parameterNumber: 3), 1)
+	]
+
+	delayBetween(cmds, 2500)
+}
+
+def ping() {
+	refresh()
+}
+
+private secureEncap(cmd, endpoint = null) {
+	response(secure(encap(cmd, endpoint)))
+}
+
+private secure(cmd) {
+	if(zwaveInfo.zw.endsWith("s")) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+def switchMode() {
+	def currentMode = device.currentValue("thermostatMode")
+	def supportedModes = state.supportedModes
+	if (supportedModes && supportedModes.size() && supportedModes[0].size() > 1) {
+		def next = { supportedModes[supportedModes.indexOf(it) + 1] ?: supportedModes[0] }
+		def nextMode = next(currentMode)
+		setThermostatMode(nextMode)
+	} else {
+		log.warn "supportedModes not defined"
+		getSupportedModes()
+	}
+}
+
+def sendEventToChild(event) {
+	String childDni = "${device.deviceNetworkId}:2"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent(event)
+}
+
+private refreshChild() {
+	def cmds = [
+			secureEncap(zwave.batteryV1.batteryGet(), 2),
+			secureEncap(zwave.sensorMultilevelV5.sensorMultilevelGet(), 2)
+	]
+	sendHubCommand(cmds, 2000)
+}
+
+private delayedRefresh() {
+	sendHubCommand(refresh())
+}
+
+def addChild() {
+	def supportedModes = []
+	supportedModes << "off"
+	supportedModes << "heat"
+	supportedModes << "auto"
+	state.supportedModes = supportedModes
+	sendEvent(name: "supportedThermostatModes", value: supportedModes, displayed: false)
+
+	String childDni = "${device.deviceNetworkId}:2"
+	String componentLabel =	 "Fibaro Temperature Sensor"
+	String ch = "ch2"
+
+	addChildDevice("Child Temperature Sensor", childDni, device.hub.id,[completedSetup: true, label: componentLabel, isComponent: false, componentName: ch, componentLabel: componentLabel])
+}
+
+private getMaxHeatingSetpointTemperature() {
+	temperatureScale == 'C' ? 30 : 86
+}
+
+private getMinHeatingSetpointTemperature() {
+	temperatureScale == 'C' ? 10 : 50
+}
+
+private fahrenheitToCelsius(temp) {
```

Comment:
`fahrenheitToCelsius` should be available to you by default, no need to define it